snow/engine/common/timer_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package common
+
+import (
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestTimeoutScheduler(t *testing.T) {
+	for _, testCase := range []struct {
+		expectedInvocationCount int
+		desc                    string
+		shouldPreempt           bool
+		clock                   chan time.Time
+		initClock               func(chan time.Time)
+		advanceTime             func(chan time.Time)
+	}{
+		{
+			desc:                    "multiple pendingTimeout one after the other with preemption",
+			expectedInvocationCount: 10,
+			shouldPreempt:           true,
+			clock:                   make(chan time.Time, 1),
+			initClock:               func(_ chan time.Time) {},
+			advanceTime:             func(_ chan time.Time) {},
```

Comment:
Super nit, but typically we don't put the `_` if we don't need to.
```suggestion
			initClock:               func(chan time.Time) {},
			advanceTime:             func(chan time.Time) {},
```